eng/tools/lint-diff/src/lint-diff.ts

```diff
@@ -0,0 +1,303 @@
+import { readFile } from "fs/promises";
+import { exec } from "node:child_process";
+import { parseArgs, ParseArgsConfig } from "node:util";
+import { getOpenapiType, getAllTags, getInputFiles } from "./markdown-parser.js";
+import { sep, dirname, join } from "path";
+import { pathExists } from "./util.js";
+
+// 64 MiB max ouptut buffers for exec
+const MAX_EXEC_BUFFER = 64 * 1024 * 1024;
+
+function usage() {
+  console.log("TODO: Write up usage");
+}
+
+export async function main() {
+  let validArgs = true;
+  const config: ParseArgsConfig = {
+    options: {
+      path: {
+        type: "string",
+        short: "p",
+      },
+      "changed-files-path": {
+        type: "string",
+        short: "c",
+      },
+      "out-file": {
+        type: "string",
+        short: "o",
+      },
+      "lint-version": {
```

Comment:
This parameter should be removed, and instead all necessary dependencies should be listed in a package.json.  If the dependencies are to be called directly during dev inner-loop, they should be listed in the root package.json.  Else, use the package,json of the tool that calls the dependencies.

Reply:
Also, `autorest` is special because, by default, it uses custom code to manage it's own dependendencies.  However, it can be convinced to use only pre-installed deps.  We will want to use this here, and I'll extract it into a common JS helper.

Here's me making this fix to `tsp-client`:

https://github.com/Azure/azure-sdk-tools/pull/8610
https://github.com/Azure/azure-sdk-tools/pull/9043

Reply:
This has been done.